### ISSUE TITLE
performance: postpone require edits to completionItem/resolve

### ISSUE THREAD

**Issue by mainej** [clojure-lsp/clojure-lsp#1068]
Sometimes completion knows that for a completion item to be valid the `ns` form will have to be edited to require a new namespace. It calculates the edit for each completion item and includes that in its response. When there are many of these completion items, the edit calculations slow down completion. These calculations could be postponed to `completionItem/resolve`, so that only one edit is calculated, and only after the user selects its item. See #1053 for background.

**Comment by mainej** [comment]
As I've been researching this I noticed that the existing code postpones documentation to `completionItem/resolve`. But it doesn't check whether the client supports that (i.e. whether "documentation" appears in the list of [`CompletionClientCapabilities.completionItem.resolveSupport.properties`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionClientCapabilities)). @ericdallo two questions:
1. Do you think that should be ignored, addressed as part of this, or made into a separate issue?
2. How should it be fixed? We can do the documentation lookup earlier, in `textDocument/completion`, or we can skip it entirely. I vote for the second, because otherwise I think we'll end up looking up documentation for every clojure.core function.

**Comment by ericdallo** [comment]
> Do you think that should be ignored, addressed as part of this, or made into a separate issue?

It should be a simple if right? I'm ok adding to the same PR.

> How should it be fixed? We can do the documentation lookup earlier, in textDocument/completion, or we can skip it entirely. I vote for the second, because otherwise I think we'll end up looking up documentation for every clojure.core function.

Yes, the second is the best IMO, just check if the capability is available and search for the docs

**Comment by mainej** [comment]
> check if the capability is available and search for the docs

👍 

> It should be a simple if right?

It is simple but requires many changes to thread the data for the condition. Still, not too hard.
